api/v1alpha1/inferencepool_types.go

```diff
@@ -59,8 +59,88 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
 	TargetPortNumber int32 `json:"targetPortNumber"`
+
+	// EndpointPickerConfig specifies the configuration needed by the proxy to discover and connect to the endpoint
+	// picker service that picks endpoints for the requests routed to this pool.
+	EndpointPickerConfig `json:"endpointPickerConfig"`
+}
+
+// EndpointPickerConfig specifies the configuration needed by the proxy to discover and connect to the endpoint picker extension.
+// This type is intended to be a union of mutually exclusive configuration options that we may add in the future.
+type EndpointPickerConfig struct {
+	// Extension configures an endpoint picker as an extension service.
+	//
+	// +kubebuilder:validation:Required
+	Extension *ExtensionConfig `json:"extension"`
+}
+
+// ExtensionConfig specifies how to configure an extension that runs the endpoint picker.
+type ExtensionConfig struct {
+	// ExtensionRef is a reference to a service extension.
+	ExtensionRef ExtensionReference `json:"extensionRef"`
+
+	// ExtensionConnection configures the connection between the gateway and the extension.
+	ExtensionConnection `json:"extensionConnection"`
 }
 
+// ExtensionReference is a reference to the extension deployment.
+type ExtensionReference struct {
```

Comment:
Missing namespace if it's namespace-scoped?

Reply:
The ref must be in the same ns as the referent, i.e. this InferencePool. It's a good idea to explicitly state this in the ExtensionReference godocs.